pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -54,6 +54,10 @@ func (r *AtlasProjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	if !result.IsOk() {
 		return result.ReconcileResult(), nil
 	}
+	if project.ConnectionSecretObjectKey() != nil {
+		r.EnsureResourcesAreWatched(req.NamespacedName, "Secret", log, *project.ConnectionSecretObjectKey())
```

Comment:
this is where the connection secret watch registration happens. If the connection secret reference has changed - this will handle removal of the previous secret and will register the new one